statistics/table.go

```diff
@@ -325,26 +325,41 @@ func (t *Table) GetRowCountByColumnRanges(sc *stmtctx.StatementContext, colID in
 func (t *Table) GetRowCountByIndexRanges(sc *stmtctx.StatementContext, idxID int64, indexRanges []*ranger.NewRange) (float64, error) {
 	idx := t.Indices[idxID]
 	if t.Pseudo || idx == nil || idx.Len() == 0 {
-		return getPseudoRowCountByIndexRanges(sc, indexRanges, float64(t.Count))
+		colsLen := -1
+		if idx != nil && idx.Info.Unique {
+			colsLen = len(idx.Info.Columns)
+		}
+		return getPseudoRowCountByIndexRanges(sc, indexRanges, float64(t.Count), colsLen)
 	}
 	result, err := idx.getRowCount(sc, indexRanges)
 	result *= idx.getIncreaseFactor(t.Count)
 	return result, errors.Trace(err)
 }
 
 // PseudoTable creates a pseudo table statistics.
-func PseudoTable(tableID int64) *Table {
-	return &Table{
-		TableID: tableID,
+func PseudoTable(tblInfo *model.TableInfo) *Table {
```

Comment:
Could we cache the pseudo table stats in TableInfo?

Reply:
Yes.